initfiles/componentfiles/configxml/sparkThor.xsl.in

```diff
@@ -0,0 +1,113 @@
+            <xsl:with-param name="val" select="@MESOS_NATIVE_JAVA_LIBRARY"/>
+        </xsl:call-template>
+        <xsl:call-template name="printVariable">
+            <xsl:with-param name="var" select="'HADOOP_CONF_DIR'"/>
+            <xsl:with-param name="val" select="@HADOOP_CONF_DIR"/>
+        </xsl:call-template>
+        <xsl:call-template name="printVariable">
+            <xsl:with-param name="var" select="'YARN_CONF_DIR'"/>
+            <xsl:with-param name="val" select="@YARN_CONF_DIR"/>
+        </xsl:call-template>
+        <xsl:call-template name="printVariable">
+            <xsl:with-param name="var" select="'SPARK_EXECUTOR_CORES'"/>
+            <xsl:with-param name="val" select="@SPARK_EXECUTOR_CORES"/>
+        </xsl:call-template>
+        <xsl:call-template name="printVariable">
+            <xsl:with-param name="var" select="'SPARK_EXECUTOR_MEMORY'"/>
+            <xsl:with-param name="val" select="@SPARK_EXECUTOR_MEMORY"/>
+        </xsl:call-template>
+        <xsl:call-template name="printVariable">
+            <xsl:with-param name="var" select="'SPARK_MASTER_HOST'"/>
+            <xsl:with-param name="val" select="@SPARK_MASTER_HOST"/>
+        </xsl:call-template>
+        <xsl:call-template name="printVariable">
+            <xsl:with-param name="var" select="'SPARK_MASTER_PORT'"/>
+            <xsl:with-param name="val" select="@SPARK_MASTER_PORT"/>
+        </xsl:call-template>
+        <xsl:call-template name="printVariable">
+            <xsl:with-param name="var" select="'SPARK_MASTER_OPTS'"/>
+            <xsl:with-param name="val" select="@SPARK_MASTER_OPTS"/>
+        </xsl:call-template>
+        <xsl:call-template name="printVariable">
+            <xsl:with-param name="var" select="'SPARK_WORKER_CORES'"/>
+            <xsl:with-param name="val" select="@SPARK_WORKER_CORES"/>
+        </xsl:call-template>
+        <xsl:call-template name="printVariable">
+            <xsl:with-param name="var" select="'SPARK_WORKER_MEMORY'"/>
+            <xsl:with-param name="val" select="@SPARK_WORKER_MEMORY"/>
+        </xsl:call-template>
+        <xsl:call-template name="printVariable">
+            <xsl:with-param name="var" select="'SPARK_WORKER_PORT'"/>
+            <xsl:with-param name="val" select="@SPARK_WORKER_PORT"/>
+        </xsl:call-template>
+        <xsl:call-template name="printVariable">
+            <xsl:with-param name="var" select="'SPARK_WORKER_OPTS'"/>
+            <xsl:with-param name="val" select="@SPARK_WORKER_OPTS"/>
+        </xsl:call-template>
+        <xsl:call-template name="printVariable">
+            <xsl:with-param name="var" select="'SPARK_DAEMON_MEMORY'"/>
+            <xsl:with-param name="val" select="@SPARK_DAEMON_MEMORY"/>
+        </xsl:call-template>
+        <xsl:text>SPARK_PID_DIR=@PID_PATH@</xsl:text>
+        <xsl:text>SPARK_CONF_DIR=@RUNTIME_PATH@/</xsl:text><xsl:value-of select="@name"/>
+    </xsl:template>
```

Comment:
can we verify that NodeGroup/@name exists in the actual /Environment/Software/ThorCluster/@name ?